docs/docsite/rst/stored_state_schema.rst

```diff
@@ -0,0 +1,113 @@
+.. _ansible_collections.cloud.pravic.docsite.stored_state:
+
+Specification format for storing resource state
+===============================================
+
+Storing resource state allows users to have a detailed view about the resources they have provisioned earlier.
+By storing resource state, a user would be able to reference their defined resources and declare a state of `absent` and expect Ansible to resolve the relationships between resources and determine the order of operations rather than requiring the user to write a properly ordered playbook.
+This resource state specification is provider-agnostic and allows some resource group metadata to be stored in a `State` dictionary.
+
+The format for storing resource state should be something like:
+
+```yaml
+
+    [{
+        "ResourceGroupName":{
+            "description":"The unique identifier for a group of resources.",
+            "type":"string"
+        },
+        "State": {
+          "description":"Dictionary of metadata for ResourceGroupName.",
+          "type":"object",
+          "properties": {
+              "Created": {
+                  "description":"Date and time in RFC3339 format that the ResourceGroup was created.",
+                  "type":"date"
+              },
+              "Modified": {
+                  "description":"Date and time in RFC3339 format that the ResourceGroup was last modified.",
+                  "type":"date"
+              },
+              "ResourceGroup": {
+                  "description":"The unique identifier for a group of resources.",
+                  "type":"string"
+              },
+              <anything else we think makes sense>
+          }
+        },
+        "Resources":{
+            "description":"Dictionary of resources belonging to ResourceGroupName.",
+            "type":"object",
+            "properties": {
```

Comment:
Same as for the resource declaration format, I think this should use `patternProperties`.